packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/AuthCognito.kt

```diff
@@ -130,9 +131,17 @@ public class AuthCognito : FlutterPlugin, ActivityAware, MethodCallHandler, Plug
 
   override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?): Boolean {
     var isHostedUIActivity = isRedirectActivityDeclared(context)
-    if (!isHostedUIActivity && requestCode == AWSCognitoAuthPlugin.WEB_UI_SIGN_IN_ACTIVITY_CODE) {
-      Amplify.Auth.handleWebUISignInResponse(data)
-      return true
+    if (requestCode == AWSCognitoAuthPlugin.WEB_UI_SIGN_IN_ACTIVITY_CODE) {
+        /// The HostedUI activity in amplify-android handles success case
+        /// We need a response handler if the HostedUI activity isn't used...
+        if (!isHostedUIActivity) {
+            Amplify.Auth.handleWebUISignInResponse(data)
+            return true
+        /// ... or if the HostedUI activity is used, but the user cancels
+        } else if (isHostedUIActivity && resultCode == RESULT_CANCELED) {
+            Amplify.Auth.handleWebUISignInResponse(data)
+            return true
+        }
```

Comment:
This could be shortened a bit:

```suggestion
         /// ... or if the HostedUI activity is used, but the user cancels
        if (!isHostedUIActivity || resultCode == RESULT_CANCELED) {
            Amplify.Auth.handleWebUISignInResponse(data)
            return true
        }
```